src/components/header/header.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { Error } from '../../types/errors';
+import { USER_ID } from '../../api/todos';
+import { TodoServiceApi } from '../../utils/todoService';
+
+type Props = {
+  areTodosActive: boolean;
+  setErrorMessage: (error: string) => void;
+  todos: Todo[];
+  setTodos: (todos: Todo[] | ((prevTodos: Todo[]) => Todo[])) => void;
+  setTempTodo: (tempTodo: Todo | null) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  areTodosActive,
+  setErrorMessage,
+  setTodos,
+  setTempTodo,
+}) => {
+  const [todoText, setTodoText] = useState<string>('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useLayoutEffect(() => {
+    inputRef.current?.focus();
+  }, []);
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    inputRef.current?.focus();
+    setIsSubmitting(true);
+
+    const trimmedTodo = todoText.trim();
+
+    if (trimmedTodo.length === 0) {
```

Comment:
Here you cancel creation of todo but `isSubmitting` is still `true`.  Add `setIsSubmitting(false)` it should help.